### PR TITLE
fix: add min/max support to DayPicker

### PR DESCRIPTION
**Pull request by smolster**
## Motivation
Two:
1. Adding a mechanism to specify a min/max date, as well as selectively disabling dates arbitrarily.
2. Allow specifying arbitrary props to the internal day picker, to allow more flexible customization.

### CODE REVIEW

File: src/components/DayPicker/DayPicker.tsx [hunk 1]
```diff
@@ -15,11 +14,13 @@ import { composeEventHandlers } from '../../utils';
...
-  date.setMinutes(0);
-  date.setSeconds(0);
-  date.setMilliseconds(0);
+const midnightOf = (date: Date) => {
+  const copy = new Date(date.toISOString());
+  copy.setHours(0);
+  copy.setMinutes(0);
+  copy.setSeconds(0);
+  copy.setMilliseconds(0);
+  return copy;
```
Comment (smolster) [inline src/components/DayPicker/DayPicker.tsx:23]: immutability is better!

**Review by danperkins — APPROVED** [review]
Looks good to me - to prevent thrashing on Chroma versions would it be worth allowing the consumer to specify arbitrary `ReactDayPicker` props and spread those?

**Comment by smolster** [comment]
@danperkins seems like a good idea. I'll target for a subsequent PR.

**Comment by smolster** [comment]
Actually, I'll just add to this PR for ease.

**Review by xdumaine — APPROVED** [review]

File: src/components/DayPicker/DayPicker.tsx (hunk 1 above)
Comment (ynotdraw) [reply src/components/DayPicker/DayPicker.tsx:26]: Great idea 👍

**Review by ynotdraw — APPROVED** [review]
Excellent work!

**Review by HaleyWardo — APPROVED** [review]

**Review by danperkins — APPROVED** [review]

**Pull request merged**